# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../kbd/devpts.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./openvt/openvt.c.orig	2003-08-16 16:44:04.000000000 +0300
+++ ./openvt/openvt.c	2003-08-16 16:42:01.000000000 +0300
@@ -100,18 +100,18 @@
    }
 
    consfd = getfd();
-   if (consfd < 0) {
-      fprintf(stderr,
-	      _("Couldnt get a file descriptor referring to the console\n"));
-      return(2);
-   }
 
-   if (ioctl(consfd, VT_GETSTATE, &vtstat) < 0) {
+   if (consfd >= 0 && ioctl(consfd, VT_GETSTATE, &vtstat) < 0) {
 	perror("openvt: VT_GETSTATE");
         return(4);
    }
 
    if (vtno == -1) {
+     if (consfd < 0) {
+        fprintf(stderr,
+	      _("Couldnt get a file descriptor referring to the console\n"));
+        return(2);
+     }
      if ((ioctl(consfd, VT_OPENQRY, &vtno) < 0) || (vtno == -1)) {
         perror("openvt: VT_OPENQRY");
 	fprintf(stderr, _("openvt: cannot find a free vt\n"));
@@ -267,7 +267,7 @@
 
    if ( do_wait ) {
       wait(NULL);
-      if (show) { /* Switch back... */
+      if (show && consfd >= 0) { /* Switch back... */
 	 if (ioctl(consfd, VT_ACTIVATE, vtstat.v_active)) {
 	    perror("VT_ACTIVATE");
 	    return 8;
